### PR TITLE
support for Vue 3.1

### PR DESCRIPTION
**Pull request by kazupon**

### CODE REVIEW

File: packages/vue-i18n/src/components/DatetimeFormat.ts [hunk 1]
```diff
@@ -3,8 +3,24 @@ import { DatetimePartsSymbol } from '../composer'
...
 import { baseFormatProps } from './base'
 import { assign } from '@intlify/shared'
+import { defineComponent } from 'vue'
 
-import type { RenderFunction, SetupContext } from 'vue'
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import type {
+  RenderFunction,
+  // NOTE: avoid https://github.com/microsoft/rushstack/issues/1050
+  // @ts-ignoreu
```
Comment (DannyFeliz) [inline packages/vue-i18n/src/components/DatetimeFormat.ts:12]:
```suggestion
  // @ts-ignore
```

File: packages/vue-i18n/src/components/NumberFormat.ts [hunk 2]
```diff
@@ -3,8 +3,24 @@ import { NumberPartsSymbol } from '../composer'
...
 import { baseFormatProps } from './base'
 import { assign } from '@intlify/shared'
+import { defineComponent } from 'vue'
 
-import type { SetupContext, RenderFunction } from 'vue'
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import type {
+  RenderFunction,
+  // NOTE: avoid https://github.com/microsoft/rushstack/issues/1050
+  // @ts-ignoreu
```
Comment (DannyFeliz) [inline packages/vue-i18n/src/components/NumberFormat.ts:12]:
```suggestion
  // @ts-ignore
```

Comment (DannyFeliz) [inline packages/vue-i18n/src/components/NumberFormat.ts:12]:
```suggestion
  // @ts-ignore
```

**Comment by kazupon** [comment]
@DannyFeliz 
Thanks! :)

**Pull request merged**